index.html

```diff
@@ -3410,6 +3410,191 @@ <h3>
         <h2>
 DID URL Dereferencing
         </h2>
+        <p>
+The DID URL dereferencing function dereferences a <a>DID URL</a> into
+a resource with contents depending on the <a>DID URL</a>'s components,
+including the <a>DID method</a>, method-specific identifier, query parameters,
+and fragment. When a <a>DID URL</a> is also a <a>DID</a>, this process
+ performs <a>DID Resolution</a>. The details of how this
+process is accomplished are outside the scope of this specification, but all
+conformant implementations MUST implement two functions which have the
+following abstract forms:
+        </p>
+
+        <p><code>
+dereference ( did-url, did-url-dereferencing-input-metadata ) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&gt; ( did-url-dereferencing-metadata, content, content-metadata )
+        </code></p>
+
+        <p>
+The input variables of these functions MUST be as follows:
+        </p>
+
+        <dl>
+            <dt>
+did-url
+            </dt>
+            <dd>
+A conformant <a>DID URL</a> as a single string. This is the <a>DID URL</a> to dereference.
+This input is REQUIRED.
+            </dd>
+            <dt>
+did-url-dereferencing-input-metadata
+            </dt>
+            <dd>
+A <a href="metadata-structure">metadata structure</a> consisting of input
+options to the <code>dereference</code> function in addition to the <code>did-url</code>
+itself.
+Properties defined by this specification are in <a href="#did-url-dereferencing-input-metadata-properties"></a>.
+This input is REQUIRED, but the structure MAY be empty.
+            </dd>
+        </dl>
+
+        <p>
+The output variables of these functions MUST be as follows:
+        </p>
+
+        <dl>
+            <dt>
+did-url-dereferencing-metadata
+            </dt>
+            <dd>
+A <a href="metadata-structure">metadata structure</a> consisting of values
+relating to the results of the <a>DID URL Dereferencing</a> process. This
+structure is REQUIRED and MUST NOT be empty.
+Properties defined by this specification are in 
+<a href="#did-url-dereferencing-metadata-properties"></a>.
+If the dereferencing is not successful, this structure MUST contain an
+<code>error</code> property describing the error.
+            </dd>
+            <dt>
+content
+            </dt>
+            <dd>
+If the <code>dereferencing</code> function was called and successful, this MUST
+contain a resource corresponding to the <a>DID URL</a>.
+Additionally, in the case that the <a>DID URL</a> is a <a>DID</a>, then this
+MUST be a <code>did-document</code> as described in <a>DID Resolution</a>,
```

Comment:
```suggestion
The content MAY be a <a>DID document</a> in one of the conformant <a href="#core-representations">.
```
Similar to one of my earlier comments, I would prefer slightly more abstract language rather than a concrete statement about what the result will be. For example, there are DID URLs which contain more than just DIDs and still return a DID document. I think we don't want so many details on what exactly the function returns in what situations (this should go into the DID Resolution spec, see [this algorithm](https://w3c-ccg.github.io/did-resolution/#dereferencing-algorithm)).